src/virtual_machine/instructions.rs

```diff
@@ -0,0 +1,246 @@
+use super::{lc3_virtual_machine::LC3VirtualMachine, register::Register, trap::Trap};
+
+pub enum Instruction {
+    BR = 0,
+    ADD,
+    LD,
+    ST,
+    JSR,
+    AND,
+    LDR,
+    STR,
+    RTI,
+    NOT,
+    LDI,
+    STI,
+    JMP,
+    RES,
+    LEA,
+    TRAP,
+}
+
+impl From<u16> for Instruction {
+    fn from(value: u16) -> Self {
+        match value {
+            0 => Instruction::BR,
+            1 => Instruction::ADD,
+            2 => Instruction::LD,
+            3 => Instruction::ST,
+            4 => Instruction::JSR,
+            5 => Instruction::AND,
+            6 => Instruction::LDR,
+            7 => Instruction::STR,
+            8 => Instruction::RTI,
+            9 => Instruction::NOT,
+            10 => Instruction::LDI,
+            11 => Instruction::STI,
+            12 => Instruction::JMP,
+            13 => Instruction::RES,
+            14 => Instruction::LEA,
+            15 => Instruction::TRAP,
+            _ => panic!("Wrong Instruction code"),
+        }
+    }
+}
+
+pub fn add(
+    virtual_machine: &mut LC3VirtualMachine,
+    destination_register: Register,
+    source_one_register: Register,
+    source_two_register: Register,
+) {
+    let new_register_value = virtual_machine.read_register(source_one_register)
```

Comment:
You'll have to use `.wrapping_add()`. If the sum of the contents of the registers overflow `u16`, the program will panic, which is not the desired behavior. This comment applies for all additions in operations.

Reply:
Oh, didn't know about that, in another branch was casting from u32 and u16 as crazy because of that